daffodil-test/src/test/resources/org/apache/daffodil/section00/general/tunables.tdml

```diff
@@ -811,4 +832,78 @@
 
   </tdml:unparserTestCase>
 
+  <tdml:defineSchema name="intPatternRestriction">
+    <xs:include schemaLocation="/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd" />
+
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited" />
+
+    <xs:element name="int">
+      <xs:simpleType>
+        <xs:restriction base="xs:int">
+          <xs:pattern value=".*[05]" /> <!-- must be a multiple of 5 -->
+        </xs:restriction>
+      </xs:simpleType>
+    </xs:element>
+
+  </tdml:defineSchema>
+
+  <tdml:parserTestCase
+    name="invalidRestrictionPolicyError_01" root="int"
+    model="intPatternRestriction"
+    config="cfg_invalidRestrictionPolicyError"
+    validation="limited">
+    <tdml:document>201</tdml:document>
+    <tdml:errors>
+      <tdml:error>Schema Definition Error</tdml:error>
+      <tdml:error>Pattern restriction</tdml:error>
+      <tdml:error>string</tdml:error>
+    </tdml:errors>
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase
+    name="invalidRestrictionPolicyIgnore_01" root="int"
+    model="intPatternRestriction"
+    config="cfg_invalidRestrictionPolicyIgnore"
+    validation="limited">
+    <tdml:document>201</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:int>201</ex:int>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:validationErrors /> <!-- there should be no validation errors -->
+  </tdml:parserTestCase>
+
+  <tdml:parserTestCase
+    name="invalidRestrictionPolicyValidate_01" root="int"
+    model="intPatternRestriction"
+    config="cfg_invalidRestrictionPolicyValidate"
+    validation="limited">
+    <tdml:document>201</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:int>201</ex:int>
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+    <tdml:validationErrors>
+      <tdml:error>failed facet checks</tdml:error>
+      <tdml:error>pattern</tdml:error>
+      <tdml:error>.*[05]</tdml:error>
+    </tdml:validationErrors>
```

Comment:
Actually, if you need to double check whether it is necessary to match validation errors, this test is easier to edit than any of the other tests.  The only change you need is to comment out lines 888 to 892 and check if the test still passes (and whether you see any of the validation messages in the test's output).

Reply:
Mentioned above, but validation errors are ignored if there is no `<tdml:validationErrors>` element, so with those lines removed this test would still pass but would not verify that validation occurred.